main.tf

```diff
@@ -5,51 +5,29 @@ terraform {
 locals {
   # VPC Net/Subnet names ---------------------------------------------------------------------------
   vpc_name               = format("%s-%s", var.name_vpc_network, var.name_suffix)
-  subnet_name_public     = format("%s-%s-%s", var.name_public_subnets, "%s", var.name_suffix)
   subnet_name_private    = format("%s-%s", var.name_private_subnet, var.name_suffix)
   subnet_name_proxy_only = format("%s-%s", var.name_proxy_only_subnet, var.name_suffix)
   # VPC IP ranges ----------------------------------------------------------------------------------
   ip_ranges = {
-    public = tolist(toset(var.ip_ranges.public))
     private = {
       primary = var.ip_ranges.private_primary
       k8s = flatten([
         for k8s_ip_ranges in var.ip_ranges.private_k8s : [
           {
             cidr = k8s_ip_ranges.pods,
-            name = (k8s_ip_ranges.name != "") ? (
-              format("%s-pods-%s", k8s_ip_ranges.name, var.name_suffix)
-              ) : (
-              format(
-                "private-k8spods%s-%s",
-                (index(var.ip_ranges.private_k8s, k8s_ip_ranges) > 0) ? (
-                  index(var.ip_ranges.private_k8s, k8s_ip_ranges) + 1
-                ) : "",
-                var.name_suffix
-              )
-            )
+            name = format("%s-k8spods-%s", coalesce(k8s_ip_ranges.name, "private"), var.name_suffix)
           },
           {
             cidr = k8s_ip_ranges.svcs,
-            name = (k8s_ip_ranges.name != "") ? (
-              format("%s-svcs-%s", k8s_ip_ranges.name, var.name_suffix)
-              ) : (
-              format(
-                "private-k8ssvcs%s-%s",
-                (index(var.ip_ranges.private_k8s, k8s_ip_ranges) > 0) ? (
-                  index(var.ip_ranges.private_k8s, k8s_ip_ranges) + 1
-                ) : "",
-                var.name_suffix
-              )
-            )
+            name = format("%s-k8ssvcs-%s", coalesce(k8s_ip_ranges.name, "private"), var.name_suffix)
```

Comment:
As a self-clarifying-message, `k8ssvcs` communicates its purposes more clearly than `svcs` which can imply any kind of services (including non-k8s ones). Anyone looking at the subnet console and its IP range names without any prior context will be clear what this is about and what this is not about. The matter of `k8s` is not actually implied here without prior context because there is no mention of `k8s` in the subnet or in its attributes

And besides, `k8ssvcs` is **needed** for backward compatibility of our existing IP range names Otherwise, it will be a destructive operation.

k8spods is needed for backward compatibility of existing IP ranges

Reply:
There are only handful usecases in Google cloud for IP range allocation, if a user is looking at IP ranges, with svc/pods, it would be clear for the user that these are for a VPC native cluster IP address range allocation.

implied doesn’t necessarily mean that we say k8s-manifests, manifests implicity gives the meaning to the user.

If it’s not backward compatible then its fine.

however what if any IaC has public subnet, wouldn’t removing public subnets from here cause destructive action too.

i actually don’t understand why this MR is needed for common-infra consolidation either.

perhaps, going forward, we should have more description in the description as to what this MR is trying to achieve. Asking to review an MR in terraform without any description is hard as it’s not always clear what functionalites are impacted.

Reply:
Agreed. That's why i shared the link to the commits tab in the review channel. The commit-by-commit review would clarify what is being done here and why. I shall copy the commit-by-commit messages into the description for clarity.